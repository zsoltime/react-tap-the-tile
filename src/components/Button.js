import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/macro';

const StyledButton = styled.button.attrs(props => ({
  type: props.type ? props.type : 'button',
}))`
  background-color: ${props =>
    props.primary ? '#46b980' : '#363636'};
  border: 0;
  border-radius: 0.25rem;
  color: ${props => (props.primary ? 'inherit' : '#fff')};
  font-size: 1.5rem;
  max-width: 12rem;
  padding: 0.75rem 1rem;
  user-select: none;
  width: 100%;

  &:focus {
    outline: thin dotted;
  }
`;

const Button = ({ children, className, onClick, ...props }) => (
  <StyledButton className={className} onClick={onClick} {...props}>
    {children}
  </StyledButton>
);

Button.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

export default Button;
