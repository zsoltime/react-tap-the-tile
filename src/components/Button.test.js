import React from 'react';
import { mount, render, shallow } from 'enzyme';
import 'jest-styled-components';

import Button from './Button';

describe('<Button />', () => {
  it('matches snapshot', () => {
    const wrapper = render(
      <Button onClick={f => f}>
        <span aria-hidden={true}>Close</span>&times;
      </Button>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot if primary', () => {
    const wrapper = render(
      <Button onClick={f => f} primary>
        <span aria-hidden={true}>Close</span>&times;
      </Button>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('passes the className prop down', () => {
    const wrapper = mount(
      <Button className="your-class" onClick={f => f}>
        Start Test
      </Button>
    );

    expect(
      wrapper.find('button').hasClass('your-class')
    ).toBeTruthy();
  });

  it('calls the callback prop on click', () => {
    const handleClick = jest.fn();
    const wrapper = shallow(
      <Button onClick={handleClick}>Start Test</Button>
    );

    wrapper.simulate('click');

    expect(handleClick).toBeCalledTimes(1);
  });
});
