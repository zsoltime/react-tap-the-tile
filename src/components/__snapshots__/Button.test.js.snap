// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
.c0 {
  background-color: #363636;
  border: 0;
  border-radius: 0.25rem;
  color: #fff;
  font-size: 1.5rem;
  padding: 0.75rem 1rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
}

.c0:focus {
  outline: thin dotted;
}

<button
  class="c0"
  type="button"
>
  <span
    aria-hidden="true"
  >
    Close
  </span>
  ×
</button>
`;

exports[`<Button /> matches snapshot if primary 1`] = `
.c0 {
  background-color: #6ab68a;
  border: 0;
  border-radius: 0.25rem;
  color: inherit;
  font-size: 1.5rem;
  padding: 0.75rem 1rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
}

.c0:focus {
  outline: thin dotted;
}

<button
  class="c0"
  type="button"
>
  <span
    aria-hidden="true"
  >
    Close
  </span>
  ×
</button>
`;
